### ISSUE TITLE
Preprocessing test text data fails

### ISSUE THREAD

**Issue by chrixrux** [amaiya/ktrain#314]
I want to use a test set in addition to the train and validation set produced by the texts_from_* methods.
But when I try to preprocess it before passing it on to evaluate I receive an error that the label encoder was not trained.

Please find sample code with the issue below. (Its tested on Colab)
```(python)
!pip3 install ktrain

import ktrain
from ktrain import text

import pandas as pd
from sklearn.datasets import fetch_20newsgroups

# fetch the dataset using scikit-learn
categories = ['alt.atheism', 'soc.religion.christian', 'comp.graphics', 'sci.med']
train_b = fetch_20newsgroups(subset='train',categories=categories, shuffle=True, random_state=42)
test_b = fetch_20newsgroups(subset='test', categories=categories, shuffle=True, random_state=42)

x_train = train_b.data
y_train = train_b.target
train_data = pd.DataFrame({"text": x_train, "label": y_train})
train_data = train_data.replace([0, 1, 2, 3],['alt.atheism', 'soc.religion.christian', 'comp.graphics', 'sci.med'])

x_test = test_b.data
y_test = test_b.target
test_data = pd.DataFrame({"text": x_test, "label": y_test})
test_data = test_data.replace([0, 1, 2, 3],['alt.atheism', 'soc.religion.christian', 'comp.graphics', 'sci.med'])

#preprocess and create 10% validation set
trn, val, preproc = text.texts_from_df(train_data,
                                       text_column="text",
                                       label_columns=["label"],
                                       preprocess_mode='standard',
                                       maxlen=350, 
                                       max_features=35000)


model = text.text_classifier('nbsvm', train_data=trn, preproc=preproc)
learner = ktrain.get_learner(model, train_data=trn, val_data=val)
learner.fit(0.001, 3, cycle_len=1, cycle_mult=2)
learner.evaluate() # Evaluation on validation set works fine

tst = preproc.preprocess_test(test_data["text"].values, test_data["label"].values) # Raises ValueError: LabelEncoder has not been trained. Call with train=True
learner.evaluate(test_data=tst)
```

The error output is:
```(python)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-6cb5b3483f34> in <module>()
     37 learner.evaluate() # Evaluation on validation set works fine
     38 
---> 39 tst = preproc.preprocess_test(test_data["text"].values, test_data["label"].values) # Error
     40 learner.evaluate(test_data=tst)

2 frames
/usr/local/lib/python3.6/dist-packages/ktrain/utils.py in apply(self, targets, train)
    583         # string targets (classification)
    584         elif len(targets.shape) == 1 and isinstance(targets[0], str):
--> 585             if not train and self.le is None: raise ValueError('LabelEncoder has not been trained. Call with train=True')
    586             if train:
    587                 self.le = LabelEncoder()

ValueError: LabelEncoder has not been trained. Call with train=True
```

**Comment by amaiya** [comment]
@chrixrux Thank you for reporting this issue and also for providing an easy way to reproduce the problem.

I found the cause of the issue (which appears to be confined to `texts_from_df`) and pushed a quick fix to `develop` branch.  It will be included in the `ktrain==0.25.4`, which should be released soon.

Thanks again.

**Comment by chrixrux** [comment]
Thanks for the quick resolution!
